### ISSUE TITLE
Fix issue with HTML comments with empty line

### ISSUE THREAD

**Issue by fletcher** [fletcher/MultiMarkdown-6#25]
An HTML comment containing an empty line is not treated as a comment, but the initial line(s) is/are still treated as HTML for block parsing.  Doesn't affect HTML output, but does affect LaTeX and others.

**Comment by fletcher** [comment]
To avoid backtracking, I suspect the only way to handle this involves two options:

1. Leave as is and treat comments like other HTML markup.  This requires leaving out blank lines.

2. Treat HTML comments like fenced code blocks -- once opened, you either close them, or they continue through the end of the document.   (This is the approach used by CommonMark)

**Comment by dgreen** [comment]
From the user side, it would seem that #2 is preferable.

**Comment by mn4367** [comment]
So at least this is partially comparable to #41?

**Comment by fletcher** [comment]
@mn4367 -- Yes...  It's basically the same problem, complicated by the fact that CriticMarkup is, by definition, "orthogonal" to Markdown.

**Comment by fletcher** [comment]
The other thing to note -- comments are treated like other HTML spans in that they can either start at the beginning of a line (HTML "Block") or inside a line (HTML "Span").

An HTML Span is parsed inside of another block (e.g. a paragraph) and must therefore be completed inside the paragraph.

An HTML Block is it's own block, so can contain empty lines.

**Comment by fletcher** [comment]
I pushed a commit to the development branch to address this.  If the `<!--` or `-->` are on lines by themselves, they can start or stop a comment block.  Otherwise, they must be part of another block (e.g. a paragraph).

The test suite has some (updated) examples to clarify this, particularly `HTML Comments`.

I also fixed an issue with non-HTML outputs where some HTML comments would leak through but others would not.  For now, HTML comments are limited to HTML only.  I want to work on this further in issue #38.
